client/engine/engine.go

```diff
@@ -159,8 +159,29 @@ func (e *Engine) handleChainEvent(chainEvent chainservice.Event) ObjectiveChange
 // Approve an existing objective (if not null)
 func (e *Engine) handleAPIEvent(apiEvent APIEvent) ObjectiveChangeEvent {
 	if apiEvent.ObjectiveToSpawn != nil {
-		return e.attemptProgress(apiEvent.ObjectiveToSpawn)
+
+		switch request := (apiEvent.ObjectiveToSpawn).(type) {
+
+		case virtualfund.ObjectiveRequest:
+			vfo, err := virtualfund.SpawnObjective(request, e.store.GetTwoPartyLedger)
+			if err != nil {
+				panic(err)
```

Comment:
I understand that we are not advised to `panic` in go. Is there a reason it's ok here?

Reply:
This is mostly due to my personal preference 😛  I much prefer panicking over ignoring an error, so it's easy to figure out test failures and debug into them.

Right now we don't really have an error handling framework so we often ignore errors in the engine. This can make debugging/tracing pretty confusing (which motivated this [PR](https://github.com/statechannels/go-nitro/pull/346)) as the errors get swallowed and things don't fail until later.

Perhaps a better fix than panicking in individual engine functions is to just bubble up errors to the root `Run` loop. The `Run` loop can `panic` on error for now (to aid in debugging) and then later can be replaced with something more sophisticated.

Reply:
That seems to point to incorrect error handling, doesn't it? I feel like we should address that by correctly handling errors, instead of panicking (unless we have a good reason to panic).

Reply:
I would argue that error handling is basically not implemented at the moment, which leads to errors being ignored or not handling consistently. We don't have to use `panic`s but I do think unhandled errors in the engine should cause things to fail immediately (unless we can handle that error gracefully) and should fail consistently.

This can be tackled outside of this PR, as I can switch `func (e *Engine) handleAPIEvent(apiEvent APIEvent)` to return an error in this PR and just return an error.

Reply:
I've removed the panics in [900370c](https://github.com/statechannels/go-nitro/pull/341/commits/900370c27ae4277a9a26271277a8551e80917c03) and created a new issue (#353) to deal with the fact we're ignoring errors in the engine.